### DISCUSSION TITLE
switch_set_on_time not working as expected

### DISCUSSION THREAD

**Discussion by BigProgrammer2000** [SukramJ/aiohomematic#1816] [Ideas]
### The problem

I have noticed that the homematicip_local.switch_set_on_time action/service call does not work as in a program in the CCU.
In the CCU, the ON_TIME parameter can be overwritten after a switch has been switched on. Unfortunately, this does not work with homematicip_local.switch_set_on_time. A reset with the value 0 according to the documentation is also not possible.

**Example:**
Call homematicip_local.switch_set_on_time with 10s
Call switch.turn_on
wait 5s
Call homematicip_local.switch_set_on_time with 100s
Call switch.turn_on

Result: The switch only stays on for 10s.
Expected result: The switch should stay on for 100s.

Same behavior when using homematicip_local.set_device_value.

Overwriting ON_TIME works when using homematicip_local.put_paramset. However, it is necessary to send “STATE: true” so that the overwriting works.
I also found out the following when testing homematicip_local.put_paramset:
```
ON_TIME: 5
STATE: true
-> switched on for 5s.

ON_TIME: 10
STATE: true
-> overwrites the previous value with 10s.

ON_TIME: -1
STATE: true
-> deletes the timer. The switch remains switched on.

ON_TIME: 0
STATE: true
-> switches the switch off immediately. Same as STATE: false.
```

**Suggestion:**
A checkbox could be added to send STATE: true in the service call. In addition, a distinction must then be made between checkbox active (homematicip_local.put_paramset) and not active (homematicip_local.set_device_value)

**Attention:**
I could not check if the same applies to homematicip_local.light_set_on_time. 

It would be nice if homematicip_local.switch_set_on_time could be adapted so that overwriting ON_TIME really works.

**Comment by SukramJ** [comment]
> does not work as in a program in the CCU.

This has never been an assumption

**Reply by BigProgrammer2000** [reply]
Sorry, that was more of an assumption on my part. 
The only thing is that the reset as described in the documentation with the value 0 does not work if the ON_TIME parameter cannot be overwritten. Apart from that. That the timer is deleted after the end anyway and also when the switch is switched off manually.

Of course, in an automation system with a motion detector, this makes sense if you always reset the value of, say, 2min as soon as motion is detected. Then the light stays on for as long as motion is detected +2min.

**Reply by SukramJ** [reply]
Which device model do you use.

**Reply by BigProgrammer2000** [reply]
I'm not quite sure what you mean, but I use the motion event from a Unifi Protect Camera as a trigger and switch on a channel from a HMIPW-DRS8 with switch_set _on_time and switch.turn_on

**Reply by SukramJ** [reply]
HMIPW-DRS8 is correct

**Reply by BigProgrammer2000** [reply]
Is it possible to add an override in the service call?

**Reply by SukramJ** [reply]
???

**Reply by BigProgrammer2000** [reply]
A checkbook or something. So that when calling switch_set_on_time the parameters on_time and state are transferred in one and so on_time can be reset and changed while the switch remains switched on.

**Reply by SukramJ** [reply]
I think that should not be necessary.

**Reply by SukramJ** [reply]
Please test beta version 1.70.0b0

**Reply by BigProgrammer2000** [reply]
Gerade erst gesehen dass man hier auch auf deutsch schreiben kann. Macht die Sache etwas leichter.
Ich habe die Beta getestet. Es Funktioniert alles so wie es sein sollte. Auch dass -1 den Timer wieder löscht ohne die Lampe auszuschalten.
Vielen Dank für die schnelle Umsetzung.

**Reply by SukramJ** [reply]
Danke für die schnelle Rückmeldung, und den Umsetzungsvorschlag.
